nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -535,6 +565,46 @@ nav_msgs::msg::Path RegulatedPurePursuitController::transformGlobalPlan(
   return transformed_plan;
 }
 
+inline double RegulatedPurePursuitController::dot(double u_x, double u_y, double v_x, double v_y)
+{
+  // Supports only 2D
+  return  (u_x * v_x) + (u_y * v_y);
+}
+
+double RegulatedPurePursuitController::detectPathOrientation(
+  const geometry_msgs::msg::PoseStamped & pose )
+{
+  double path_direction_changed = 0.0;
+  double cusp_distance = 0.0;
+
+  // Iterating through the global path to determine the position of the cusp
+  for (auto iter = 1; iter <= (int)global_plan_.poses.size(); ++iter) {
+    // We have two vectors for the dot product OA and AB. Determining the vectors.
+    double oa_x = global_plan_.poses[iter].pose.position.x -
+      global_plan_.poses[iter - 1].pose.position.x;
+    double oa_y = global_plan_.poses[iter].pose.position.y -
+      global_plan_.poses[iter - 1].pose.position.y;
+    double ab_x = global_plan_.poses[iter + 1].pose.position.x -
+      global_plan_.poses[iter].pose.position.x;
+    double ab_y = global_plan_.poses[iter + 1].pose.position.y -
+      global_plan_.poses[iter].pose.position.y;
+
+    // Checking for the existance of cusp using the dot product
+    path_direction_changed = dot(oa_x, oa_y, ab_x, ab_y) > 0.0 ? 1.0 : -1.0;
+
+    // Getting the distance of the cusp, if the cusp is found
+    if (path_direction_changed == -1.0) {
+      auto x = global_plan_.poses[iter].pose.position.x - pose.pose.position.x;
+      auto y = global_plan_.poses[iter].pose.position.y - pose.pose.position.y;
+      cusp_distance = sqrt((x * x) + (y * y));
```

Comment:
This could be `return hypot(...)` and then the return at the bottom of this function could be `return 0.0` and that would eliminate the `cusp_distance` variable